app/Controller/Component/RoleComponent.php

```diff
@@ -1108,7 +1202,42 @@ public function isCoOrCouAdmin($coPersonId, $coId=null) {
     
     return $this->cachedGroupCheck($coPersonId, "", "", null, false, GroupEnum::Admins, true);
   }
-  
+
+  /**
+   * Determine if a CO Person is a CO or COU Approver.
+   *
+   * @since  COmanage Registry v4.2.0
+   * @param  Integer CO Person ID
+   * @param  Integer CO ID that CO Person is an Approver for, or null for any CO
+   * @return Boolean True if the CO Person is a CO or COU Approver, false otherwise
+   */
+
+  public function isCoOrCouApprover($coPersonId, $coId=null) {
+    // A person is a CO Approver if they are a member of the GroupEnum::Approvers group for the specified CO.
+    // A person is a COU Approver if they are a member of the GroupEnum::Approvers group within the specified COU.
+
+    global $group_sep;
+
+    if($coId) {
+      // First check that $coPersonId is in $coId
+
+      if(!$this->isCoPerson($coPersonId, $coId)) {
+        return false;
+      }
+    }
+
+    // For code readability, we do this as separate checks rather than passing an OR
+    // condition to cachedGroupCheck(). This may result in two DB calls, but it may not
+    // since chances are we've already cached the results to isCoApprover() (if we're being
+    // called from CoEnrollmentFlow::authorize(), at least).
+
+    if($this->cachedGroupCheck($coPersonId, "", "", null, false, GroupEnum::Approvers)) {
+      return true;
+    }
+
+    return $this->cachedGroupCheck($coPersonId, "", "", null, false, GroupEnum::Approvers, true);
```

Comment:
Shouldn't the last parameter here be `$couId` (which needs to be passed to this function)? (The comment for `cachedGroupCheck()` appears to be inconsistent with the actual function signature.)

Reply:
ok, right, the old code is a mess to read...